### ISSUE TITLE
All glusterd2 instances need not be etcd servers

### ISSUE THREAD

**Issue by prashanthpai** [gluster/glusterd2#209]
Currently, every glusterd instance is an etcd server which participates in RAFT consensus. This is undesirable for large clusters and there should only be a limited odd number of glusterd2 instances running fully-fledged etcd servers. Other nodes should be etcd proxies.

Currently, etcd does not provide an embed like API to start embedded etcd in proxy mode. However, the etcd proxy can be embedded in glusterd2 similar to how [etcd internally does it](https://github.com/coreos/etcd/blob/v3.1.0-rc.1/etcdmain/etcd.go#L210). A setback there is, the current proxy implementation of etcd only supports v2 API.

**Comment by purpleidea** [comment]
FYI: This is done automatically with mgmt and can be used via libmgmt.
https://github.com/gluster/glusterd2/pull/197

**Comment by prashanthpai** [comment]
@purpleidea Right. But AFAIK libmgmt doesn't do etcd proxy yet (only client). Anyways, I'm adding github issues for TODOs and features so that we don't forget about it.

**Comment by purpleidea** [comment]
On Fri, Jan 6, 2017 at 7:58 AM, Prashanth Pai <notifications@github.com> wrote:
>
> @purpleidea Right. But AFAIK libmgmt doesn't do etcd proxy yet (only client). Anyways, I'm adding github issues for TODOs and features so that we don't forget about it.

You won't need the proxy feature since this magic is done automatically in mgmt.

**Comment by kshlm** [comment]
If not #261 this needs to happen.
